src/backend/gl/src/factory.rs

```diff
@@ -92,42 +107,67 @@ impl Factory {
     fn create_buffer_internal(&mut self) -> Buffer {
         let gl = &self.share.context;
         let mut name = 0 as Buffer;
-        unsafe {
-            gl.GenBuffers(1, &mut name);
-        }
+        unsafe { gl.GenBuffers(1, &mut name); }
         info!("\tCreated buffer {}", name);
         name
     }
 
-    fn init_buffer(&mut self, buffer: Buffer, info: &f::BufferInfo) {
+    fn init_buffer(&mut self,
+                   buffer: Buffer,
+                   info: &f::BufferInfo,
+                   data: Option<&[u8]>) {
+        use gfx_core::factory::Usage::*;
+
         let gl = &self.share.context;
         let target = role_to_target(info.role);
+        let data = if let Some(data) = data {
```

Comment:
oh my, double override of the same variable name
Please use different names to avoid confusion.